pinot-segment-local/src/main/java/org/apache/pinot/segment/local/dedup/BasePartitionDedupMetadataManager.java

```diff
@@ -30,22 +31,27 @@
 import org.apache.pinot.common.metrics.ServerTimer;
 import org.apache.pinot.segment.local.indexsegment.immutable.EmptyIndexSegment;
 import org.apache.pinot.segment.local.indexsegment.immutable.ImmutableSegmentImpl;
+import org.apache.pinot.segment.local.utils.WatermarkUtils;
 import org.apache.pinot.segment.spi.IndexSegment;
+import org.apache.pinot.segment.spi.V1Constants;
 import org.apache.pinot.spi.config.table.HashFunction;
 import org.apache.pinot.spi.data.readers.PrimaryKey;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 public abstract class BasePartitionDedupMetadataManager implements PartitionDedupMetadataManager {
+  // The special value to indicate the largest seen time is not set yet, assuming times are positive.
+  protected static final double TTL_WATERMARK_NOT_SET = 0;
```

Comment:
Consider making it consistent with upsert (using `-Inf`)?

Reply:
as dedup TTL is done with dedupTime a time column, so it'd be safe to assume positive time values
but it sounds good to me to make it consistent with upsert comparison column. will try in another pr.